### DISCUSSION TITLE
Modbus RTU (RS485) a prerequisite to operate the PID

### DISCUSSION THREAD

**Discussion by BjorkmanMartin** [artisan-roaster-scope/artisan#785] [Q&A] [answered]
Hi -
Thanks for reading and supporting an all different subjects. I've seen in the posts that quite some energy is put into this.

To my question:
It is said in the early documentation that Modbus RTU / RS485 is a prerequisite to make use of the built in PID controller.
In the configuration menu, Modbus TCP is an option.
Will the built in PID controller function also if TCP is choosen?

Grateful for advise :-)

Kind regards
Martin

**Comment by MAKOMO** [comment, marked as answer]
Not sure which "early documentation" you refer to. The [blog post on the PID support](https://artisan-roasterscope.blogspot.com/2016/11/pid-control.html) states that you need to specify a PID slaveID (in the MODBUS tab) to control an external PID via MODBUS. Just don't specify a PID slaveID (keep it at the default 0) and the internal software PID will be used.

**Comment by BjorkmanMartin** [comment]
Hi Marko,
I caught the sentence when browsing through the text, but did not put it in the right context. Sorry to bother about nothing ...
To create structure for myself, I organized info as per enclosed.

![Screenshot 2022-01-06 191656](https://user-images.githubusercontent.com/97189793/148431572-d961a704-fb58-4433-bd4e-1e7815146b41.png)

Kind regards
Martin

**Reply by MAKOMO** [reply]
Dear Martin, that graph is about it. Thus I would describe it a bit differently. The PID support in Artisan has two parts, the PID Dialog (the UI) and the actual PID algorithm. There are roughly your 3 cases

- external PID algorithm with some specific support (Fuji, Delta, Arduino). Note that with the Fuji/Delta modules Artisan is communicate via MODBUS, while the Arduino (assumes the aArtisan PID firmware) comes with its own serial protocol.
- external PID algorithm connected via MODBUS, but without specific support. One specifies the SV, p, i, d registers to be set and the slaveID of the external module
- internal Software PID which is the only Artisan internal PID loop

All variants come with a similar, but not identical UI (PID Dialog) opened via the Control button. The internal PID works with anything that Artisan can control, but has the drawback over the external PID loops that it runs only at the sample loop speed, while the external PID loops run asynchronously at higher speeds.

So in summary, there is not one PID support in Artisan but quite some flavours of it. Hopefully enough for most applications.
